test/Conversion/amd/dedup-by-constancy.mlir

```diff
@@ -0,0 +1,30 @@
+// RUN: triton-opt %s -split-input-file --convert-triton-amdgpu-to-llvm=arch=gfx942 --convert-builtin-func-to-llvm | FileCheck %s
+
+// CHECK-LABEL: dedup_by_constancy_mfma
+// CHECK-COUNT-4: llvm.icmp "slt"
+// CHECK-NOT: llvm.icmp "slt"
+// Here is why we expect exactly 4 icmp:
+// For a 32x32 tensor A with mfma layout, each thread holds 16 elements, which are divided
+// into 4 groups. E.g. thread 0 holds elements A[0:3,0], A[8:11,0], A[16:19,0], and A[24:27,0].
+// In this example, constancy of the tensor is 16 for dim 0, meaning A[0:15,0] have same values
+// and A[16:31,0] have same values. Therefore, for thread 0, the first 8 elements are duplicated
+// and the last 8 elements are duplicated. Ideally, thread 0 only needs two icmp, one for the
+// first 8 elements and the other for the last 8 elements. In practice, the dedup analysis
+// only allows duplication within each group of 4 elemnets. Therefore, we expect 4 icmp, one
```

Comment:
Could you please explain why the dedup analysis only allow duplication within each group of 4 elements?

Reply:
It's explained in the comment here: https://github.com/triton-lang/triton/blob/938b5ca91217d15f32bdaac5bc4ead1d0ee94615/include/triton/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVMBase.h#L119-L120

In the case that constancy=8 and contigPerThread=4, we cannot duplicate 8 elements within each thread, since the second group of 4 elements of the thread goes beyond the constancy.